backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/KeyspaceFactory.java

```diff
@@ -33,10 +44,33 @@ public static void createKeyspace(ClusterConfiguration clusterConfiguration, Clu
 
     private static void doCreateKeyspace(ClusterConfiguration clusterConfiguration, Cluster cluster) {
         try (Session session = cluster.connect()) {
-            session.execute("CREATE KEYSPACE IF NOT EXISTS " + clusterConfiguration.getKeyspace()
-                + " WITH replication = {'class':'SimpleStrategy', 'replication_factor':" + clusterConfiguration.getReplicationFactor() + "}"
-                + " AND durable_writes = " + clusterConfiguration.isDurableWrites()
-                + ";");
+            if (!keyspaceExist(cluster, clusterConfiguration.getKeyspace())) {
+                session.execute(SchemaBuilder.createKeyspace(clusterConfiguration.getKeyspace())
+                    .with()
+                    .replication(ImmutableMap.<String, Object>builder()
+                        .put("class", "SimpleStrategy")
+                        .put("replication_factor", clusterConfiguration.getReplicationFactor())
+                        .build())
+                    .durableWrites(clusterConfiguration.isDurableWrites()));
+            }
+        }
+    }
+
+    @VisibleForTesting
+    public static boolean keyspaceExist(Cluster cluster, String keyspaceName) {
+        try (Session session = cluster.connect(SYSTEM_SCHEMA)) {
+            long numberOfKeyspaces = session.execute(select()
+                    .countAll()
+                    .from(KEYSPACES)
+                    .where(eq(KEYSPACE_NAME, keyspaceName)))
+                .one()
+                .getLong("count");
+
+            if (numberOfKeyspaces > 1 || numberOfKeyspaces < 0) {
+                throw new IllegalStateException(String.format("unexpected keyspace('%s') count being %d", keyspaceName, numberOfKeyspaces));
+            }
```

Comment:
is it useful?

Reply:
I feel uncomfortable when dropping this check. I would like to keep it if you don't mind!